Add package name to NotifyReporter notification

GeorgeSapkin opened the pull request:
## Summary

This change adds package name to notification in NotifyReporter when using `--notify` flag or `notify` configuration option. This clarifies which package is being tested which is especially useful when watching tests in mono-repos and with multiple IDEs opened.

First I'm trying to get hasteFS from the first context to get the module name (must be a better way). Then I'm falling back to `rootDir` and otherwise to original behavior without an additional title. 

Before:

![Before](https://user-images.githubusercontent.com/5013206/38090619-71194b18-3363-11e8-861e-4ca5566ea6a3.png)

After:

![After](https://user-images.githubusercontent.com/5013206/38090620-71388e10-3363-11e8-88fa-65e8ca605ef0.png)

SimenB commented on packages/jest-cli/src/reporters/notify_reporter.js:
```diff
@@ -22,6 +23,8 @@ const isDarwin = process.platform === 'darwin';
 
 const icon = path.resolve(__dirname, '../assets/jest_logo.png');
 
+const readFile = util.promisify(fs.readFile);
```
Not available in node 6, please use pify

SimenB replied:
Disregard this, shouldn't need fs here at all?

SimenB commented on packages/jest-cli/src/reporters/notify_reporter.js:
```diff
@@ -37,10 +40,21 @@ export default class NotifyReporter extends BaseReporter {
+      'package.json',
+    );
+    const {name: packageName = this._globalConfig.rootDir} = JSON.parse(
+      await readFile(packagePath),
```
This file should be available in hastefs already, should not be necessary to hit the file system

GeorgeSapkin replied:
I'll check it out.

rickhanlonii commented:
This sounds great! Do you have a screenshot of the new notification?

GeorgeSapkin commented:
@SimenB I've rewritten the code a bit using hastefs. I don't think this change has anything to do with `test-node-9` failing in CI.

@rickhanlonii I've updated the PR description with screenshots.

rickhanlonii commented:
And yeah, the node 9 failure is unrelated 👌

SimenB commented:
Sorry about the silence, forgot about this one!

If you rebase on master, the test failure should go away.

This needs a test as well :)

rickhanlonii commented:
@GeorgeSapkin bump, this is great, would love to get this in for Jest 23

GeorgeSapkin commented:
@SimenB I've rebased and existing checks are passing. What kind of additional tests do I need to write?

SimenB commented:
There are no tests added in this PR, some unit tests asserting that the newly added code does its job would be awesome.
You can mock out `node-notifier` to verify it's called with the correct arguments

GeorgeSapkin commented:
I'll try to add the missing tests this weekend.

thymikee commented:
@GeorgeSapkin mind rebasing this? After that we should be good to merge, sorry for this taking so long!

GeorgeSapkin commented:
`deploy/netlify` is failing. How can I address that?

thymikee commented:
It's failing allover the repo, I don't know what's the issue :(. 
cc @endiliey
